golang/cosmos/e2e_test/util.go

```diff
@@ -0,0 +1,271 @@
+	}
+
+	ret.ChainConfig.ChainID = chainID
+	return ret
+}
+
+// newAgoricChainSpec fully specifies the details necessary to launch an Agoric chain image built from
+// - https://github.com/strangelove-ventures/heighliner/pull/211
+// - NoCrisisModule is a flag added in the agoric-labs fork of interchaintest
+func newAgoricChainSpec(chainUniqueName string, chainID string, chainImage ibc.DockerImage, numOfValidators int, numOfFullNodes int) *interchaintest.ChainSpec {
+	coinDecimals := int64(6)
+	gasAdjustment := 1.3
+	noHostMount := false
+
+	return &interchaintest.ChainSpec{
+		Name:          "agoric",
+		ChainName:     chainUniqueName,
+		Version:       chainImage.Version,
+		GasAdjustment: &gasAdjustment,
+		NoHostMount:   &noHostMount,
+		ChainConfig: ibc.ChainConfig{
+			Type:    "cosmos",
+			Name:    "agoric",
+			ChainID: chainID,
+			Images: []ibc.DockerImage{
+				chainImage,
+			},
+			Bin:          "agd",
+			Bech32Prefix: "agoric",
+			Denom:        "ubld",
+			CoinType:     "564",
+			// interchaintest is super flaky when gas is enabled
+			GasPrices:      "0.0ubld",
+			GasAdjustment:  1.3,
+			TrustingPeriod: "672h",
+			NoHostMount:    false,
+			NoCrisisModule: true,
+			SkipGenTx:      false,
+			CoinDecimals:   &coinDecimals,
+		},
+		NumValidators: &numOfValidators,
+		NumFullNodes:  &numOfFullNodes,
+	}
+}
+
+// getChainImage will build a docker image from the environment variable value
+// E2ETEST_CHAINIMAGE_AGORIC. The value of this env var
+// must be in the form "repo/image:version"
+func getChainImageAgoric(t *testing.T) ibc.DockerImage {
+	ret := ibc.DockerImage{
+		UidGid: "1025:1025",
+	}
+
+	chainImage, present := os.LookupEnv(ENV_CHAINIMAGE_AGORIC)
+	if !present {
+		chainImage = DEFAULT_CHAINIMAGE_AGORIC
+	}
+
+	parts := strings.Split(chainImage, ":")
+	if len(parts) == 2 {
+		ret.Repository = parts[0]
+		ret.Version = parts[1]
+	} else {
+		t.Fatalf("Invalid value for %s[%s]. Must be of the format 'repository:version'", ENV_CHAINIMAGE_AGORIC, chainImage)
+	}
+
+	t.Logf("ChainImages: %s[%s:%s]", ENV_CHAINIMAGE_AGORIC, ret.Repository, ret.Version)
+
+	return ret
+}
+
+// getChainNames reads the environment variables FMT_ENV_CHAINNAME0, FMT_ENV_CHAINNAME1, FMT_ENV_CHAINNAME2, FMT_ENV_CHAINNAME3
+func getChainNames(t *testing.T) [4]string {
+
+	ret := [4]string{
+		CHAIN_AGORIC, CHAIN_AGORIC, CHAIN_AGORIC, CHAIN_AGORIC,
+	}
+
+	for i := 0; i < 4; i++ {
+		envVar := fmt.Sprintf(FMT_ENV_CHAINNAME, i)
+		chainName, present := os.LookupEnv(envVar)
+		if present {
+			ret[i] = chainName
+		}
+	}
+
+	t.Logf("ChainNames: %s[%s] %s[%s] %s[%s] %s[%s]",
+		fmt.Sprintf(FMT_ENV_CHAINNAME, 0), ret[0],
+		fmt.Sprintf(FMT_ENV_CHAINNAME, 1), ret[1],
+		fmt.Sprintf(FMT_ENV_CHAINNAME, 2), ret[2],
+		fmt.Sprintf(FMT_ENV_CHAINNAME, 3), ret[3])
+
+	return ret
+}
+
+// getChainSpec reads environment variables and builds a full ChainSpec
+func getChainSpec(t *testing.T) []*interchaintest.ChainSpec {
+	nv := 1
+	nf := 0
```

Comment:
Consider changing `nv` to `numValidators`, `nf` to `numFull`.

Reply:
This code is idiomatic and follows https://go.dev/wiki/CodeReviewComments#variable-names